### PR TITLE
handle flight mode changes via control panel

### PR DESCRIPTION
**Pull request by danicaerediano**

### CODE REVIEW

**Comment by pmigueld** [comment]
@danicaerediano  Can you resolve conflicts?

File: screens/Scan/ScanScreenController.ts [hunk 1]
```diff
@@ -31,20 +34,22 @@ export function useScanScreen({ navigation }: MainRouteProps) {
...
+    } else {
+      if (isLocationDisabled) {
+        locationError.message = t('errors.locationDisabled.message');
+        locationError.button = t('errors.locationDisabled.button');
+      } else if (isLocationDenied) {
+        locationError.message = t('errors.locationDenied.message');
+        locationError.button = t('errors.locationDenied.button');
+      }
+    }
+
```
Comment (pmigueld) [inline screens/Scan/ScanScreenController.ts:52]: Moving it here means the location error will no longer update after the first render. Is this what you intended?

File: screens/Scan/ScanScreenController.ts [hunk 2]
```diff
@@ -66,6 +71,16 @@ export function useScanScreen({ navigation }: MainRouteProps) {
...
 
+  useEffect(() => {
+    SystemSetting.isAirplaneEnabled().then((enable) => {
+      if (enable) {
+        scanService.send(ScanEvents.FLIGHT_ENABLED());
+      } else {
+        scanService.send(ScanEvents.FLIGHT_DISABLED());
+      }
+    });
+  }, [isActive]);
```
Comment (pmigueld) [inline screens/Scan/ScanScreenController.ts:82]: Have you tried using the `APP_ACTIVE` event in the state machine? It fires from the parent app machine when the focus changes.

Comment (pmigueld) [inline screens/Scan/ScanScreenController.ts:82]: Have you tried using the `APP_ACTIVE` event in the state machine? It fires from the parent app machine when the focus changes.

File: screens/Scan/ScanScreenController.ts (hunk 1 above)
Comment (danicaerediano) [reply screens/Scan/ScanScreenController.ts:52]: @pmigueld I moved this because it gives lint error

**Review by pmigueld — APPROVED** [review]
LGTM 👍 

Verified working on Redmi 9C (MIUI Global 12.0.21, Android 10)

**Pull request merged**